proxy/server.py

```diff
@@ -44,768 +44,1023 @@
-            else:
-                message = '{"error": "TEDAPI not enabled"}'
-        elif self.path.startswith('/cloud'):
-            # Cloud API Specific Calls
-            if pw.cloudmode and not pw.fleetapi:
-                message = '{"error": "Use /cloud/battery, /cloud/power, /cloud/config"}'
-                if self.path == '/cloud/battery':
-                    message = json.dumps(pw.client.get_battery())
-                if self.path == '/cloud/power':
-                    message = json.dumps(pw.client.get_site_power())
-                if self.path == '/cloud/config':
-                    message = json.dumps(pw.client.get_site_config())
-            else:
-                message = '{"error": "Cloud API not enabled"}'
-        elif self.path.startswith('/fleetapi'):
-            # FleetAPI Specific Calls
-            if pw.fleetapi:
-                message = '{"error": "Use /fleetapi/info, /fleetapi/status"}'
-                if self.path == '/fleetapi/info':
-                    message = json.dumps(pw.client.get_site_info())
-                if self.path == '/fleetapi/status':
-                    message = json.dumps(pw.client.get_live_status())
-            else:
-                message = '{"error": "FleetAPI not enabled"}'
-        elif self.path in DISABLED:
-            # Disabled API Calls
-            message = '{"status": "404 Response - API Disabled"}'
-        elif self.path in ALLOWLIST:
-            # Allowed API Calls - Proxy to Powerwall
-            message: str = pw.poll(self.path, jsonformat=True)
-        elif self.path.startswith('/control/reserve'):
+        """Handle GET requests."""
+        path = self.path
+
+        # Map paths to handler functions
+        path_handlers = {
+            '/aggregates': self.handle_aggregates,
+            '/api/meters/aggregates': self.handle_aggregates,
+            '/soe': self.handle_soe,
+            '/api/system_status/soe': self.handle_soe_scaled,
+            '/api/system_status/grid_status': self.handle_grid_status,
+            '/csv': self.handle_csv,
+            '/vitals': self.handle_vitals,
+            '/strings': self.handle_strings,
+            '/stats': self.handle_stats,
+            '/stats/clear': self.handle_stats_clear,
+            '/temps': self.handle_temps,
+            '/temps/pw': self.handle_temps_pw,
+            '/alerts': self.handle_alerts,
+            '/alerts/pw': self.handle_alerts_pw,
+            '/freq': self.handle_freq,
+            '/pod': self.handle_pod,
+            '/version': self.handle_version,
+            '/help': self.handle_help,
+            '/api/troubleshooting/problems': self.handle_problems,
+        }
+
+        result: str = ""
+        if path in path_handlers:
+            result = path_handlers[path]()
+        elif path in DISABLED:
+            result = self.send_json_response(
+                {"status": "404 Response - API Disabled"},
+                status_code=HTTPStatus.NOT_FOUND
```

Comment:
Status code needs to be 200 for this API. The Powerflow animation code looks for it. The payload can be as it shows.

Reply:
That's a bit counterintuitive that `DISABLED` still returns 200. However, I think I sorta understand. It's not NOT FOUND in the proxy itself, it's simply disabled for use in when being called THROUGH the proxy.